### DISCUSSION TITLE
Julia Client - Read Only Option

### DISCUSSION THREAD

**Discussion by HiramTheHero** [duckdb/duckdb#14342] [Q&A]
Hello all,

I hope you've been well. Is there a "read only" mode for the Julia package? (Similar to what the R package is able to do.) I couldn't find anything within the documentation.

Thank you for your help!

**Comment by paulmelis** [comment]
Was looking for the same thing, and took me a while to figure out. It's already possible in the current Julia package:

```
julia> config = DuckDB.Config()
DuckDB.Config(Ptr{Nothing} @0x0000000012b56b50)

julia> DuckDB.set_config(config, "access_mode", "READ_ONLY")

julia> db = DuckDB.DB("test.db", config)
DuckDB.DB("test.db")

julia> DBInterface.execute(db, "create table test (v integer not null)")
ERROR: Execute of query "create table test (v integer not null)" failed: Invalid Input Error: Cannot execute statement of type "CREATE" on database "test" which is attached in read-only mode!
```

**Comment by paulmelis** [comment]
Also relevant, this pull request from a few weeks ago: https://github.com/duckdb/duckdb/pull/17585
